docs/delorean/add_new_product.md

```diff
@@ -0,0 +1,168 @@
+               wipe-workspace: false
+```
+
+The following parameters need to be set:
+
+* productName - Set to the new product name (3scale)
+* productVersionVar - The name of the variable that should be updated in the installer manifest
+
+### Implement discovery.yaml
+
+Overview of the branch jobs function can be found [here](overview.md#Branch)
+
+Copy and modify one of the existing ga/discovery.yaml files, or create one with the following:
+
+```---
+   - job:
+       name: delorean-<product-name>/ga/discovery
+       display-name: '<product-name>-ga-discovery'
+       project-type: pipeline
+       concurrent: false
+       triggers:
+         - timed: '@daily'
+       parameters:
+         - string:
+             name: 'productVersionVar'
+             default: 'UPDATE ME'
+             description: '[REQUIRED] The manifest variable to be used as the current component version'
+             read-only: true
+         - string:
+             name: 'projectOrg'
+             default: 'UDPATE ME'
+             description: '[REQUIRED] github project organization'
+             read-only: true
+         - string:
+             name: 'projectRepo'
+             default: 'UPDATE ME'
+             description: '[REQUIRED] github project repository'
+             read-only: true
+         - string:
+             name: 'productName'
+             default: 'UPDATE ME'
+             description: '[REQUIRED] Product to check, this affects the way the job verifies if a new version is available'
+             read-only: true
+       pipeline-scm:
+         script-path: jobs/delorean/jenkinsfiles/discovery/ga/github/Jenkinsfile
+         scm:
+           - git:
+               branches:
+                 - master
+               url: 'https://github.com/integr8ly/ci-cd.git'
+               skip-tag: true
+               wipe-workspace: false
+
+```
+
+The following parameters need to be set:
+
+* productName - Set to the new product name (3scale)
+* productVersionVar - The name of the variable that should be updated in the installer manifest
+* projectOrg - The github org where the product templates repo is 
+* projectRepo - The GitHub repo where the product templates are stored
+
+By default the discovery job will check for GitHub releases, but if the new product doesn't use releases,
+you can use the following optional variables to check tags instead.
+
+```      
+ - string:
+     name: 'releaseFetchMethod'
+     default: 'tag'
+     description: '[REQUIRED] Method to fetch latest release which can either be by tag or by release'
+     read-only: true
+ - string:
+     name: 'gaReleaseTagRef'
+     default: 'GA'
+     description: '[REQUIRED] Reference used to filter GA releases from the repository tags'
+     read-only: true
+```
+
+
+* releaseFetchMethod - Set to tag to tell it to check repo tags instead of the default GH releases
+* gaReleaseTagRef - Name to check in tag to indicate that it's a GA tag. Tag might look like 1.1.1.GA for example.
+
```

Comment:
Do we want to document the option for having a separate/different release tag and a product version? https://github.com/integr8ly/ci-cd/blob/master/jobs/delorean/gitea/ga/discovery.yaml#L11-L19 This would also need to explain how to get the product version using [productVersionLocation](https://github.com/integr8ly/ci-cd/blob/master/jobs/delorean/gitea/ga/discovery.yaml#L36-L44) and probably add a filter for it [here](https://github.com/integr8ly/ci-cd/blob/master/jobs/delorean/jenkinsfiles/discovery/ga/github/Jenkinsfile#L104-L135) though,

Reply:
Started adding this, but was starting to get confusing. Might have another go at it later.

Reply:
Updated this to list all parameters in a table and mark each as required or not. https://github.com/mikenairn/ci-cd/blob/INTLY-2045_add_new_product_docs/docs/delorean/add-new-product.md